apps/email/js/sync.js

```diff
@@ -75,6 +75,8 @@ define(function(require) {
           });
         }
 
+        title = title || mozL10n.get('new-message-no-subject');
```

Comment:
Lets call it notification-no-subject so that the key indicates where/how it is used, and update the LOCALIZATION NOTE comment to mention that this is used for notifications.